pkg/alertmanager/alertmanager.go

```diff
@@ -110,28 +120,31 @@ func New(cfg *Config) (*Alertmanager, error) {
 		am.wg.Done()
 	}()
 
-	marker := types.NewMarker()
-	am.alerts, err = mem.NewAlerts(marker, 30*time.Minute)
+	marker := types.NewMarker(nil)
+	am.alerts, err = mem.NewAlerts(context.Background(), marker, 30*time.Minute, am.logger)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create alerts: %v", err)
 	}
 
-	am.api = api.New(
-		am.alerts,
-		am.silences,
-		func(matchers []*labels.Matcher) dispatch.AlertOverview {
-			return am.dispatcher.Groups(matchers)
+	am.api, err = api.New(api.Options{
+		Alerts:     am.alerts,
+		Silences:   am.silences,
+		StatusFunc: marker.Status,
+		Peer:       cfg.Peer,
+		Logger:     log.With(am.logger, "component", "api"),
+		GroupFunc: func(f1 func(*dispatch.Route) bool, f2 func(*types.Alert, time.Time) bool) (dispatch.AlertGroups, map[model.Fingerprint][]string) {
+			return am.dispatcher.Groups(f1, f2)
 		},
-		marker.Status,
-		cfg.Peer,
-		log.With(am.logger, "component", "api"),
-	)
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create api: %v", err)
+	}
 
 	am.router = route.New()
 
 	webReload := make(chan chan error)
 	ui.Register(am.router.WithPrefix(am.cfg.ExternalURL.Path), webReload, log.With(am.logger, "component", "ui"))
-	am.api.Register(am.router.WithPrefix(path.Join(am.cfg.ExternalURL.Path, "/api/v1")))
+	am.api.Register(am.router, path.Join(am.cfg.ExternalURL.Path, "/api/v1"))
```

Comment:
This `Register` will handle registering the correct api prefix, no need to pass this.

```suggestion
	am.api.Register(am.router, am.cfg.ExternalURL.Path)
```